File: DIYHomeAutomationAPI/DIYHomeAutomationAPI/Models/RegistrationRequest.cs
using System.ComponentModel.DataAnnotations;
using System.Data;

namespace DIYHomeAutomationAPI.Models
{
    public class RegistrationRequest
    {
        [Required]
        public string Name { get; set; } = null!;

        [Required]
        public string Email { get; set; } = null!;

        [Required]
        public string UserName { get; set; } = null!;

        [Required]
        public string Password { get; set; } = null!;
    }
}

File: DIYHomeAutomation/app/src/main/java/com/example/diyhomeautomation/models/RegistrationRequest.kt
package com.example.diyhomeautomation.models

/**
 * Data class representing registration requests.
 *
 * @property name The user's name.
 * @property email The user's email address.
 * @property userName The user's chosen username.
 * @property password The user's password.
 *
 * @constructor Creates an instance of RegistrationRequest with the provided properties.
 */
data class RegistrationRequest (
    var name: String,
    var email: String,
    var userName: String,
    var password: String
)